### PR TITLE
Base guest functionalities

### LINKED ISSUE #9: Base guest functionalities
After #8, #12  and #7 implement REST endpoint for guest users.

Guest should be able to query for satellite data:
 - Query should be possible to be done by: 
     - id
     - satellite's code
 - data should consist of:
     - name
     - code
     - id
     - satellite’s current position
     - who made the satellite
     - country of origin
     - purpose
     - available actions

### PR DESCRIPTION
**Pull request by jpitucha**
resolves #9

### CODE REVIEW

File: src/routes/local.ts [hunk 1]
```diff
@@ -1,13 +1,46 @@
...
 
 router.get('/two-line-elements/:id', (req, res) => {
-    if (req.params.id == undefined) return res.sendStatus(400)
+    if (req.params.id === undefined) return res.sendStatus(400)
     n2yo.getTwoLineElements(req.params.id)
         .then(result => res.json(result))
         .catch(() => res.sendStatus(400))
 })
 
+router.get('/satellite', (req, res) => {
```
Comment (ixcviw7bw) [inline src/routes/local.ts]: This is a GET route, but your handler reads from `req.body`. GET requests have no body [0]. You probably meant to read query parameters. You can use `req.query` for that.

[0] GET requests can technically be sent with body, but browsers typically don't allow it.

**Review by ixcviw7bw — CHANGES_REQUESTED** [review]

File: src/routes/local.ts [hunk 2]
```diff
@@ -1,13 +1,46 @@
...
+    if (req.body.id === undefined && req.body.name === undefined) return res.sendStatus(400)
+    if (req.body.id) {
+        SatelliteUtilities.getSatelliteById(req.body.id)
+            .then((doc) => {
```
Comment (Jordan141) [inline src/routes/local.ts]: Give doc a more descriptive name, maybe satellite or satelliteDoc in this instance, do it for all instances of doc.

File: src/utils/satelliteUtils.ts [hunk 3]
```diff
@@ -1,17 +1,25 @@
...
+        return Satellite.findById(id).exec();
+    }
+
+    static getSatelliteByName(name: string): Promise<SatelliteInterface | null> {
+        return Satellite.findOne({ officialName: name }).exec()
     }
 
     static getSateliteCount(): Promise<number> {
-        return Satellite.countDocuments()
+        return Satellite.countDocuments().exec()
```
Comment (Jordan141) [inline src/utils/satelliteUtils.ts:19]: Remember that in large databases, esimatedDocumentCount is a significantly more performant option as it doesn't try to count the entire document list.  However, in this case countDocuments is fine, just keep it in mind.

File: src/utils/satelliteUtils.ts [hunk 4]
```diff
@@ -1,17 +1,25 @@
...
-    static async createSatelite(newSatelite: Record<string, string>): Promise<void> {
-        await Satellite.create(newSatelite)
-            .catch((err) => {
-                console.log(err)
-                throw 'error occured while putting new satellite'
-            })
-        return Promise.resolve()
+    static async createSatelite(newSatelite: SatelliteInterface): Promise<void> {
+        try {
+            await Satellite.create(newSatelite)
```
Comment (Jordan141) [inline src/utils/satelliteUtils.ts]: as much as I prefer async/await, this project is about you learning promises, stick with them.

**Review by Jordan141 — CHANGES_REQUESTED** [review]

File: src/routes/local.ts [hunk 5]
```diff
@@ -1,13 +1,46 @@
...
+                if (!doc) return res.sendStatus(400)
+                return res.json(_.pick(doc, propsToPick))
+            })
+            .catch(() => { return res.sendStatus(400) })
```
Comment (michaelKurowski) [inline src/routes/local.ts:23]: It's a stylistic thing and depends on the taste,
but you can write this thing as `.catch(() => res.sendStatus(400))`

Comment (michaelKurowski) [reply src/routes/local.ts:23]: In general, if you immediately want to return value in arrow (aka lambda) functions, you can omit curly bracers and the `return` keyword.
So `() => { return x }` becomes `() => x`. If there's only one arguments, you can also ommit parenthesis that wraps argument section so `(myArg) => {...}` becomes `myArg => {...}`.
Combining these two principles can create some neat and clean code i.e.
`(a) => {return a * 2}` can become `a => a * 2`

File: src/routes/local.ts [hunk 6]
```diff
@@ -1,13 +1,46 @@
...
 router.get('/two-line-elements/:id', (req, res) => {
-    if (req.params.id == undefined) return res.sendStatus(400)
+    if (req.params.id === undefined) return res.sendStatus(400)
     n2yo.getTwoLineElements(req.params.id)
         .then(result => res.json(result))
         .catch(() => res.sendStatus(400))
 })
 
+router.get('/satellite', (req, res) => {
+    if (req.body.id === undefined && req.body.name === undefined) return res.sendStatus(400)
```
Comment (michaelKurowski) [inline src/routes/local.ts]: If you'll receive `NaN`, `null`, `0` or `''` as `.name` or `.id`, you will end up with pending http request, never responding to it.

File: src/routes/local.ts [hunk 7]
```diff
@@ -1,13 +1,46 @@
 import express from 'express'
 import n2yo from './../api/n2yo/index.js'
+import SatelliteUtilities from '../utils/satelliteUtils'
+import _ from 'lodash'
 
 const router = express.Router()
+const propsToPick = ['_id', 'officialName', 'perigee', 'apogee', 'operator', 'operatorCountry', 'purpose']
```
Comment (michaelKurowski) [inline src/routes/local.ts]: Give it a more descriptive name.

Comment (michaelKurowski) [reply src/routes/local.ts]: Also it's a primitive constant it should be `WRITTEN_IN_THIS_MANNER`.

File: src/routes/local.ts (hunk 6 above)
Comment (michaelKurowski) [reply src/routes/local.ts]: Also, you use lodash, so you can use `_.isUndefined(x)`. It won't fix the issues i've pointed out, it's more like a tip for the future.

File: src/utils/satelliteUtils.ts [hunk 8]
```diff
@@ -1,17 +1,25 @@
...
+        } catch (err) {
+            throw 'error occured while putting new satellite'
```
Comment (michaelKurowski) [inline src/utils/satelliteUtils.ts]: Why do you do this, do you want to hide the original error? If you'll get an error from `createSatelite`, you'll still get unhandled promise rejection.

File: src/utils/satelliteUtils.ts [hunk 9]
```diff
@@ -1,17 +1,25 @@
...
+        }
+    }
+
+    static getSatelliteById(id: string): Promise<SatelliteInterface | null> {
+        return Satellite.findById(id).exec();
+    }
+
+    static getSatelliteByName(name: string): Promise<SatelliteInterface | null> {
```
Comment (michaelKurowski) [inline src/utils/satelliteUtils.ts]: Is it the `SatelliteInterface`? Isn't that a mongoose document?

File: src/routes/local.ts [hunk 10]
```diff
@@ -1,13 +1,46 @@
...
+            .then((doc) => {
+                if (!doc) return res.sendStatus(400)
+                return res.json(_.pick(doc, propsToPick))
+            })
+            .catch(() => { return res.sendStatus(400) })
+    }
+})
+
+router.get('/satellite/:name', (req, res) => {
+    if (req.params.name === undefined) return res.sendStatus(400)
```
Comment (michaelKurowski) [inline src/routes/local.ts]: If you'll get  `NaN`, `null`, `0` or `''` as `params.name`, you'll end up with pending http request.

File: src/routes/local.ts [hunk 11]
```diff
@@ -1,13 +1,46 @@
...
+    if (req.body.id === undefined && req.body.name === undefined) return res.sendStatus(400)
+    if (req.body.id) {
+        SatelliteUtilities.getSatelliteById(req.body.id)
+            .then((doc) => {
+                if (!doc) return res.sendStatus(400)
+                return res.json(_.pick(doc, propsToPick))
+            })
+            .catch(() => { return res.sendStatus(400) })
+    }
+    if (req.body.name) {
```
Comment (michaelKurowski) [inline src/routes/local.ts]: If you'll get  `NaN`, `null`, `0` or `''` as `req.body.name`, you'll end up with pending http request.

**Comment by michaelKurowski** [comment]
Can you query satellites by codes in your PR, or only by names?

**Comment by jpitucha** [comment]
I can query them both by ID and the name @michaelKurowski

**Review by Jordan141 — APPROVED** [review]

**Review by ixcviw7bw — APPROVED** [review]

**Review by michaelKurowski — APPROVED** [review]

**Pull request merged**